pkg/cmd/status/status.go

```diff
@@ -0,0 +1,616 @@
+				  }
+				}
+			  }
+		  }
+	  }
+	}`
+	assignmentsQ := `assignee:@me state:open%s%s`
+	requestedQ := `state:open review-requested:@me%s%s`
+
+	orgFilter := ""
+	if s.Org != "" {
+		orgFilter = " org:" + s.Org
+	}
+	excludeFilter := ""
+	if s.Exclude != "" {
+		for _, repo := range strings.Split(s.Exclude, ",") {
+			excludeFilter += " -repo:" + repo
+		}
+	}
+	assignmentsQ = fmt.Sprintf(assignmentsQ, orgFilter, excludeFilter)
+	requestedQ = fmt.Sprintf(requestedQ, orgFilter, excludeFilter)
+
+	return fmt.Sprintf(q, assignmentsQ, requestedQ)
+}
+
+// Populate .AssignedPRs, .AssignedIssues, .ReviewRequests
+func (s *StatusGetter) LoadSearchResults() error {
+	q := s.buildSearchQuery()
+	c := api.NewClientFromHTTP(s.Client)
+
+	var resp struct {
+		Assignments struct {
+			Edges []struct {
+				Node SearchResult
+			}
+		}
+		ReviewRequested struct {
+			Edges []struct {
+				Node SearchResult
+			}
+		}
+	}
+	err := c.GraphQL(ghinstance.Default(), q, nil, &resp)
+	if err != nil {
+		return fmt.Errorf("could not search for assignments: %w", err)
+	}
+
+	prs := []SearchResult{}
+	issues := []SearchResult{}
+	reviewRequested := []SearchResult{}
+
+	for _, e := range resp.Assignments.Edges {
+		if e.Node.Type == "Issue" {
+			issues = append(issues, e.Node)
+		} else if e.Node.Type == "PullRequest" {
+			prs = append(prs, e.Node)
+		} else {
+			panic("you shouldn't be here")
+		}
+	}
+
+	for _, e := range resp.ReviewRequested.Edges {
+		reviewRequested = append(reviewRequested, e.Node)
+	}
+
+	sort.Sort(Results(issues))
+	sort.Sort(Results(prs))
+	sort.Sort(Results(reviewRequested))
+
+	s.AssignedIssues = []StatusItem{}
+	s.AssignedPRs = []StatusItem{}
+	s.ReviewRequests = []StatusItem{}
+
+	for _, i := range issues {
+		s.AssignedIssues = append(s.AssignedIssues, StatusItem{
+			Repository: i.Repository.NameWithOwner,
+			Identifier: fmt.Sprintf("%s#%d", i.Repository.NameWithOwner, i.Number),
+			preview:    i.Title,
+		})
+	}
+
+	for _, pr := range prs {
+		s.AssignedPRs = append(s.AssignedPRs, StatusItem{
+			Repository: pr.Repository.NameWithOwner,
+			Identifier: fmt.Sprintf("%s#%d", pr.Repository.NameWithOwner, pr.Number),
+			preview:    pr.Title,
+		})
+	}
+
+	for _, r := range reviewRequested {
+		s.ReviewRequests = append(s.ReviewRequests, StatusItem{
+			Repository: r.Repository.NameWithOwner,
+			Identifier: fmt.Sprintf("%s#%d", r.Repository.NameWithOwner, r.Number),
+			preview:    r.Title,
+		})
+	}
+
+	return nil
+}
+
+// Populate .RepoActivity
+func (s *StatusGetter) LoadEvents() error {
+	perPage := 100
+	c := api.NewClientFromHTTP(s.Client)
+	query := url.Values{}
+	query.Add("per_page", fmt.Sprintf("%d", perPage))
+
+	currentUsername, err := s.CurrentUsername()
+	if err != nil {
+		return err
+	}
+
+	var events []Event
+	var resp []Event
+	pages := 2
+	for page := 1; page <= pages; page++ {
+		query.Add("page", fmt.Sprintf("%d", page))
+		p := fmt.Sprintf("users/%s/received_events?%s", currentUsername, query.Encode())
```

Comment:
How does the "events" API at `/users/USER/received_events` compare to the Notifications API at `/notifications`? They do seem to produce different results, but I don't get the semantic difference. If we were able to use the latter, we wouldn't have to query the CurrentUsername beforehand. Not a big deal though, since CurrentUsername is cached.

Reply:
It's very, very confusing; I tried to use notifications for this instead of events but had to give up. The notifications API does not include enough information to determine what is going on -- only that "there has been a notification" and then an issue or PR number. I can't deduce _what_ the notification is about unless I pull the associated data object and figure it out myself. The results back from the notifications API are also surprisingly volatile and changed in ways I couldn't quite figure out with just a few minutes between calls.